docs/usage/MTCGoClient.md

```diff
@@ -0,0 +1,113 @@
+
+replace k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.0.0-20181213153335-0fe22c71c476
+
+//openshift deps pinning
+replace github.com/openshift/api => github.com/openshift/api v0.0.0-20190716152234-9ea19f9dd578
+
+require (
+	cloud.google.com/go/storage v1.12.0 // indirect
+	github.com/Azure/azure-sdk-for-go v49.0.0+incompatible // indirect
+	github.com/Azure/go-autorest/autorest v0.11.13 // indirect
+	github.com/Azure/go-autorest/autorest/adal v0.9.8 // indirect
+	github.com/aws/aws-sdk-go v1.36.2 // indirect
+	github.com/evanphx/json-patch v4.9.0+incompatible // indirect
+	github.com/google/gofuzz v1.2.0 // indirect
+	github.com/konveyor/mig-controller v0.0.0-00010101000000-000000000000
+	github.com/matttproud/golang_protobuf_extensions v1.0.2-0.20181231171920-c182affec369 // indirect
+	github.com/onsi/gomega v1.10.3 // indirect
+	github.com/openshift/api v0.0.0-20201019163320-c6a5ec25f267 // indirect
+	github.com/prometheus/client_golang v1.8.0 // indirect
+	golang.org/x/crypto v0.0.0-20201203163018-be400aefbc4c // indirect
+	golang.org/x/net v0.0.0-20201202161906-c7110b5ffcbb // indirect
+	golang.org/x/oauth2 v0.0.0-20201203001011-0b49973bad19 // indirect
+	golang.org/x/time v0.0.0-20200630173020-3af7569d3a1e // indirect
+	google.golang.org/api v0.36.0 // indirect
+	k8s.io/api v0.17.4
+	k8s.io/utils v0.0.0-20201110183641-67b214c5f920 // indirect
+	sigs.k8s.io/controller-runtime v0.1.11
+)
+```
+4. Now let's create a `main.go` source file. This file will contain the code to communicate and interact with the Kubernetes/OpenShift cluster to perform operations on MTC resources. The following sample code snippet imports the MTC custom resources and the client sends a create `MigCluster` request to the cluster configured.
+```
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	kapi "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+func main() {
+
+	cluster := &v1alpha1.MigCluster{
+		Spec: v1alpha1.MigClusterSpec{
+			URL: "sample-url",
+			ServiceAccountSecretRef: &kapi.ObjectReference{
+				Namespace: "sample-ns",
+				Name:      "sample-migcluster",
+			},
+			Insecure: true,
+		},
+	}
+
+	cfg, err := config.GetConfig()
+	if err != nil {
+		fmt.Printf("\nerror: %v\n", err)
+		os.Exit(1)
+	}
+
+	mgr, err := manager.New(cfg, manager.Options{})
+	if err != nil {
+		fmt.Printf("\nerror: %v\n", err)
+		os.Exit(2)
+	}
+
+	client := mgr.GetClient()
+
+	err = client.Create(context.TODO(), cluster)
+	if err != nil {
+		fmt.Printf("\nerror: %v\n", err)
+		os.Exit(2)
+	}
+
+}
+```
+5. Just liek the above example, you can use the Go Client library to interact with the cluster and perform CRUD operations and much more on the MTC custom resources.
+
+**Note:** For more details and examples please refer [Go Client examples](https://github.com/kubernetes/client-go/tree/master/examples).
```

Comment:
This is using the controller runtime client not client-go. 

I would think that this would be a better doc page to link to: https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.2/pkg/client#pkg-examples

Reply:
Done, updated the PR.